visualization/app/codeCharta/ui/codeMap/threeViewer/threeViewerService.ts

```diff
@@ -33,6 +57,53 @@ export class ThreeViewerService {
 		window.addEventListener("focusout", event => this.onFocusOut(event))
 	}
 
+	initStats = (canvasElement: Element) => {
+		// TODO refactoring is needed
+		const { stats } = this
+		this.xPanel = stats.addPanel( new Stats.Panel( 'triangles', '#ff8', '#221' ) )
+		this.yPanel = stats.addPanel( new Stats.Panel( 'calls', '#f8f', '#212' ) )
+		stats.showPanel( 3 )
+
+		stats.domElement.style.position = 'absolute'
+		stats.domElement.style.left = '0'
+		stats.domElement.style.top = '0'
+		canvasElement.appendChild( stats.dom )
+	}
+
+	private updateStats = () => {
+		let webGLInfo : WebGLInfo["render"]
+		if (this.enableFXAA) {
+			webGLInfo = this.threeRendererService.composer.getInfo()
+		} else {
+			webGLInfo = this.threeRendererService.renderer.info.render
+		}
+		
+		const triangles : number = webGLInfo.triangles
+		this.maxXPanel = Math.max(this.maxXPanel,triangles)
+		this.xPanel.update( triangles, this.maxXPanel*1.3 )
+
+		const calls : number = webGLInfo.calls
+		this.maxYPanel = Math.max(this.maxYPanel,calls)
+		this.yPanel.update( calls, this.maxYPanel*1.3 )
+
+		this.stats.update()
+	}
+
+	// TODO render pass needs to be injected
+	private initComposer = () => { 
+		const pixelRatio = this.threeRendererService.renderer.getPixelRatio()
+
+		this.threeRendererService.composer.setSize( window.innerWidth* pixelRatio, window.innerHeight* pixelRatio)
+		var renderPass = new RenderPass( this.threeSceneService.scene, this.threeCameraService.camera )
+		this.threeRendererService.composer.addPass( renderPass )
```

Comment:
Idea for the refactoring: We could add a Subscriber for `onSceneChanged` and `onCameraChanged` or anything like that in the `threeRenderService` to listen for changes and to update the `RenderPass` accordingly. I am not sure, if this will work (just an idea).